File: frontend/src/services/order.ts
import { authService } from "./auth";
import { api } from "./api";

export interface OrderItem {
  id: string;
  product_id: string;
  product_title: string;
  quantity: number;
  price_snapshot: number;
}

export interface Order {
  id: string;
  user_id: string;
  subtotal: number;
  shipping_fee: number;
  tax: number;
  total_price: number;
  payment_method: string | null;
  notes: string | null;
  status: string;
  created_at: string;
  items: OrderItem[];
}

export interface OrderListResponse {
  items: Order[];
  total: number;
  page: number;
  per_page: number;
}

export interface CreateOrderPayload {
  shipping_address_id: string;
  payment_method: string;
  notes?: string;
}

function getToken(): string {
  const t = authService.getToken();
  if (!t) throw new Error("Not authenticated");
  return t;
}

export const orderService = {
  async create(payload: CreateOrderPayload): Promise<Order> {
    return api.post<Order>("/orders/create", payload, getToken());
  },

  async get(id: string): Promise<Order> {
    return api.get<Order>(`/orders/${id}`, getToken());
  },

  async my(page: number = 1, per_page: number = 10): Promise<OrderListResponse> {
    return api.get<OrderListResponse>(`/orders/my?page=${page}&per_page=${per_page}`, getToken());
  },

  async all(page: number = 1, per_page: number = 20): Promise<OrderListResponse> {
    return api.get<OrderListResponse>(`/orders/all?page=${page}&per_page=${per_page}`, getToken());
  },

  async updateStatus(order_id: string, status: string): Promise<Order> {
    return api.put<Order>(`/orders/status/${order_id}`, { status }, getToken());
  },
};

File: backend/app/schemas/order.py
import uuid
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class OrderItemResponse(BaseModel):
    id: uuid.UUID
    product_id: uuid.UUID | None = None
    product_title: str
    quantity: int
    price_snapshot: float

    class Config:
        from_attributes = True


class CreateOrderRequest(BaseModel):
    shipping_address_id: str
    payment_method: str
    notes: Optional[str] = None


class OrderResponse(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID
    subtotal: float
    shipping_fee: float
    tax: float
    total_price: float
    payment_method: str | None
    notes: str | None
    status: str
    created_at: datetime
    items: list[OrderItemResponse]


class OrderListResponse(BaseModel):
    items: list[OrderResponse]
    total: int
    page: int
    per_page: int


class StatusUpdateRequest(BaseModel):
    status: str
